### PR TITLE
solved

### PR DESCRIPTION
**Pull request by AbdulghaniAlhazmi**

### CODE REVIEW

File: sloution.kts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        return result  // return the area result
+    }
+
+}
+
+
+class Triangle() : Shape() {  //defining sub Class
+
+    override val shapeName = "Triangle" // override variable for shape name
+    override val shapeColor = "Yellow" // override variable for shape color
```
Comment (shuruqt1000) [inline sloution.kts]: why used override ? fix your comment

File: sloution.kts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    fun info(): String {  // defining info function
+        return "Shape Name: $shapeName\nShape Color: $shapeColor"  //return the name and color for shape
+    }
+
+    open fun computeArea(): Double { // defining computeArea Function
+        val num1: Double = 0.0  // variable for num1
+        val num2: Double = 0.0  // variable for num2
+        var result: Double = 0.0  // variable for area result
+        return result  // return the area result
+    }
```
Comment (shuruqt1000) [inline sloution.kts:41]: why did you make it open explain what this code dose ?
don't restate the sentences

File: sloution.kts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        result = (height * width) // compute Area for rectangle
+        return result  // return the rectangle area
+    }
+
+}
+
+class Circle() : Shape() {  //defining sub Class
+
+    override val shapeName = "Circle"  // override variable for shape name
+    override val shapeColor = "Green" // override variable for shape color
```
Comment (shuruqt1000) [inline sloution.kts]: why override these values where do they go?

File: sloution.kts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        result = (height * base) / 2  // compute Area for triangle
+        return result   // return the triangle area
+    }
+
+}
+
+class Rectangle() : Shape() {  //defining sub Class
+
+    override val shapeName = "Rectangle" // override variable for shape name
+    override val shapeColor = "Blue" // override variable for shape color
```
Comment (shuruqt1000) [inline sloution.kts]: why use over ride please explain your code

File: sloution.kts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    open val shapeName = ""  // variable for shape name
+    open val shapeColor = ""  // variable for shape name
+    open var result: Double = 0.0  // variable for area result
+    fun info(): String {  // defining info function
+        return "Shape Name: $shapeName\nShape Color: $shapeColor"  //return the name and color for shape
+    }
+
+    open fun computeArea(): Double { // defining computeArea Function
+        val num1: Double = 0.0  // variable for num1
+        val num2: Double = 0.0  // variable for num2
```
Comment (shuruqt1000) [inline sloution.kts:38]: are theses values being passed to other functions? if yes where and why ?

File: sloution.kts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+
+open class Shape {     //defining super Class
+
+    open val shapeName = ""  // variable for shape name
+    open val shapeColor = ""  // variable for shape name
+    open var result: Double = 0.0  // variable for area result
+    fun info(): String {  // defining info function
+        return "Shape Name: $shapeName\nShape Color: $shapeColor"  //return the name and color for shape
```
Comment (shuruqt1000) [inline sloution.kts:33]: explain what you did here

**Review by shuruqt1000 — CHANGES_REQUESTED** [review]

File: sloution.kts (hunk 5 above)
Comment (AbdulghaniAlhazmi) [reply sloution.kts:38]: these values not passed i made them because i get error when i override empty function

File: sloution.kts (hunk 4 above)
Comment (AbdulghaniAlhazmi) [reply sloution.kts]: override is used to change the values for shapeName and color for each sub class so we can return the shape and color it

File: sloution.kts (hunk 3 above)
Comment (AbdulghaniAlhazmi) [reply sloution.kts]: these values used in info function to return color shape and shape name

File: sloution.kts (hunk 2 above)
Comment (AbdulghaniAlhazmi) [reply sloution.kts:41]: this shape function is open so we can use inheritance and the class have 3 variable for shape name , color , result
open variable used to change the values for name and color val variable , for the result i was just trying to check if its will work

info function will return the shape name and shape color for the class when its called from other class
compute area will compute area when its called form other class

File: sloution.kts (hunk 1 above)
Comment (AbdulghaniAlhazmi) [reply sloution.kts]: its the same as rectangle and circle in the previous comment

**Comment by shuruqt1000** [comment]
great useful comments. !!
completed

**Pull request closed without merging**